src/qt/applicationbase/undomanager.cpp

```diff
@@ -174,23 +165,24 @@ class AutoSaver {
     std::thread saver_;
 };
 
-UndoManager::UndoManager(InviwoMainWindow* mainWindow)
-    : mainWindow_(mainWindow)
-    , manager_{mainWindow_->getInviwoApplication()->getWorkspaceManager()}
-    , refPath_{filesystem::findBasePath()}
-    , autoSaver_{std::make_unique<AutoSaver>(mainWindow_->getInviwoApplication())}
+UndoManager::UndoManager(InviwoApplication* app, QObject* mainWindow, int numRestoreFiles,
```

Comment:
I would pass just the Network, and the Workspace Manger instead.

Reply:
Done